packages/babel-parser/src/parser/error.js

```diff
@@ -78,7 +78,7 @@ export default class ParserError extends CommentsParser {
     else if (pos === this.state.lastTokStart) loc = this.state.lastTokStartLoc;
     else if (pos === this.state.end) loc = this.state.endLoc;
     else if (pos === this.state.lastTokEnd) loc = this.state.lastTokEndLoc;
-    else loc = getLineInfo(this.input, pos);
+    else loc = getLineInfo(this.state.startLoc.line, this.input, pos);
```

Comment:
`this.state.startLoc` will always be updated when a new AST node is parsed. We can make `getLineInfo` a parser method and then read `startLine` from parser options.